commit a4c3caee6d3819b81c8ba0e74be2dbee07a5ad2b (refs/changes/00/3300/1)
Author: Marsell Kukuljevic <marsell@joyent.com>
Date:   2018-02-06T22:12:53+13:00 (1 year, 8 months ago)
    
    TRITON-108 SAPI dies when files missing in /sapi/sapi_manifests

diff --git a/lib/server/stor/moray_local.js b/lib/server/stor/moray_local.js
index 6d65a47..b97e31b 100644
--- a/lib/server/stor/moray_local.js
+++ b/lib/server/stor/moray_local.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright (c) 2017, Joyent, Inc.
+ * Copyright (c) 2018, Joyent, Inc.
  */
 
 /*
@@ -195,10 +195,14 @@ function loadAppObjects(uuid, stor, cb) {
             vasync.forEachParallel({
                 'inputs': muuids,
                 'func': fManifests
-            }, function (err, res) {
-                _.manifests = res.successes.map(function (m) {
+            }, function vasyncCb(err, res) {
+                _.manifests = res.successes.filter(function filterCb(m) {
+                    // if a file isn't found
+                    return (m !== null);
+                }).map(function mapCb(m) {
                     return (m.value);
                 });
+
                 return (subcb(err));
             });
         }
